libraries/Microsoft.Bot.Builder/IStorage.cs

```diff
@@ -91,5 +92,25 @@ public static class StorageExtensions
 
             return values;
         }
+
+        /// <summary>
+        /// If the <paramref name="value"/> is an <see cref="IStoreItem"/> or <see cref="JObject"/>
+        /// then the value of the "ETag" property is returned.  Otherwise, null.
+        /// </summary>
+        /// <param name="value">The object to check for an ETag.</param>
+        /// <returns>The ETag property of the object passed in, if present.</returns>
+        public static string GetETagOrNull(object value)
```

Comment:
What is this for? Why does it need to be static? seems to be used on BlobStorage and CosmosDbPersistedStorage. Should it be part of the IStorage contract? Can it be internal or private to avoid expanding the surface area?